auditors/capabilities/capabilities.go

```diff
@@ -12,53 +12,51 @@ import (
-	"FSETID",           // Don’t clear set-user-ID and set-group-ID permission bits when a file is modified
-	"KILL",             // Bypass permission checks for sending signals
-	"MKNOD",            // Create special files using mknod(2)
-	"NET_BIND_SERVICE", // Bind a socket to internet domain privileged ports (port numbers less than 1024)
-	"NET_RAW",          // Use RAW and PACKET sockets
-	"SETFCAP",          // Set file capabilities
-	"SETGID",           // Make arbitrary manipulations of process GIDs and supplementary GID list
-	"SETPCAP",          // Modify process capabilities.
-	"SETUID",           // Make arbitrary manipulations of process UIDs
-	"SYS_CHROOT",       // Use chroot(2), change root directory
-}
-
 const (
 	// CapabilityAdded occurs when a capability is in the capability add list of a container's security context
 	CapabilityAdded = "CapabilityAdded"
-	// CapabilityNotDropped occurs when a capability that should be dropped is not in the capability drop list of a container's security context
-	CapabilityNotDropped = "CapabilityNotDropped"
+	// CapabilityShouldDropAll occurs when there's a drop list instead of having drop "ALL"
+	CapabilityShouldDropAll = "CapabilityShouldDropAll"
+	// CapabilityOrSecurityContextMissing  occurs when either the Security Context or Capabilities are not specified
+	CapabilityOrSecurityContextMissing = "CapabilityOrSecurityContextMissing"
 )
 
 const overrideLabelPrefix = "allow-capability-"
 
+var DefaultDropList = []string{"ALL"}
+
+var DefaultAddList = []string{""}
+
 // Capabilities implements Auditable
 type Capabilities struct {
-	dropList []string
+	addList []string
 }
 
 func New(config Config) *Capabilities {
 	return &Capabilities{
-		dropList: config.GetDropList(),
+		addList: config.GetAddList(),
 	}
 }
 
-// Audit checks that bad capabilities are dropped and no capabilities are added
+// Audit checks that bad capabilities are dropped with ALL and no capabilities are added
 func (a *Capabilities) Audit(resource k8stypes.Resource, _ []k8stypes.Resource) ([]*kubeaudit.AuditResult, error) {
 	var auditResults []*kubeaudit.AuditResult
 
 	for _, container := range k8s.GetContainers(resource) {
-		for _, cap := range mergeCapabilities(a.dropList, container) {
-			for _, auditResult := range auditContainerForCapability(container, cap, a.dropList) {
-				auditResult = override.ApplyOverride(auditResult, container.Name, resource, getOverrideLabel(cap))
+		var capabilitiesToDrop []string
+
+		for _, capability := range mergeCapabilities(container) {
+			for _, auditResult := range auditContainer(container, capability, a.addList) {
+				auditResult = override.ApplyOverride(auditResult, container.Name, resource, getOverrideLabel(capability))
+				if auditResult != nil && auditResult.Name == CapabilityShouldDropAll {
+					capabilitiesToDrop = append(capabilitiesToDrop, capability)
+					if len(capabilitiesToDrop) == 1 {
+						auditResults = append(auditResults, auditResult)
+					}
```

Comment:
The original idea behind calling `auditContainer` per capability is because we wanted to have an audit result per capability per container. But since this is not the case for the `CapabilityShouldDropAll` result (we only want one), we could move that check to a separate function (eg. `auditContainerForDropAll`) and call that outside the loop (rather than doing the work for each capability and discarding all but one). I wonder if we event need to merge capabilities anymore, I think we could just use the add list?

```go
for _, container := range k8s.GetContainers(resource) {
  for _, capability := range GetCapabilityDropList(container) {
    // call auditContainer and add to audit results
  }

  auditResult := auditContainerForDropAll(container)
  // add audit result to results if not nil, don't need to check for override label
}
```